problems/test_bc/test_bc.cpp

```diff
@@ -0,0 +1,93 @@
+/*! \file test_faspfenics.cpp
+ *
+ *  \brief Main to test FASP/FENICS interface using the Poisson problem
+ *
+ *  \note Currently initializes the problem based on specification
+ */
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <dolfin.h>
+#include <vector>
+#include "boundary_conditions.h"
+#include "Poisson.h"
+
+using namespace dolfin;
+//using namespace std;
+
+// Source term (right-hand side)
+class Source : public Expression
+{
+  void eval(Array<double>& values, const Array<double>& x) const
+  {
+    double dx = x[0] - 0.5;
+    double dy = x[1] - 0.5;
+    values[0] = 10*exp(-(dx*dx + dy*dy) / 0.02);
+    values[1] = 10*exp(-(dx*dx + dy*dy) / 0.02);
+    values[2] = 10*exp(-(dx*dx + dy*dy) / 0.02);
+  }
+};
+
+// Normal derivative (Neumann boundary condition)
+class dUdN : public Expression
+{
+  void eval(Array<double>& values, const Array<double>& x) const
+  {
+    values[0] = sin(5*x[0]);
+    values[1] = sin(5*x[1]);
+    values[2] = sin(5*x[2]);
+  }
+};
+
+int main()
+{
+
+  parameters["linear_algebra_backend"] = "Eigen"; // or uBLAS
+  parameters["allow_extrapolation"] = true;
+
+  // Create mesh and function space
+  dolfin::Point p0( -10.0, -10.0, -10.0);
+  dolfin::Point p1(  10.0,  10.0,  10.0);
+  dolfin::BoxMesh mesh(p0, p1, 10, 10, 5);
+
+  Poisson::FunctionSpace V(mesh);
+
+  // Define boundary condition
+  Constant ux(0.0);
+  XBoundaries bdary_x(10.0);
```

Comment:
What do you think if we simplify this to read in a list?
Something like:
```
unsigned int coordinate_index_array[4] = {0,0,2,2};
double coordinate_value_array[4] = {-L_x,L_x,L_z,L_y};
int direction_array[4] = {-1,1,-1,1};
Boundaries bndry( 
  &coordinate_index_array,
  &coordinate_value_array,
  &direction_array
);
```
I think the arrays are self-explanatory except ``direction_array``, which determines whether we are setting a lower bound (``-1``) or an upper bound (``+1``).

What do you think?  The function would return an array of ``DirichletBS``s.